src/login/logind-dbus.c

```diff
@@ -1434,23 +1435,38 @@ static int method_activate_session_on_seat(sd_bus_message *message, void *userda
         if (r < 0)
                 return r;
 
-        r = manager_get_session_from_creds(m, message, session_name, error, &session);
-        if (r < 0)
-                return r;
+        if (streq(session_name, none_id)) {
+                r = manager_get_seat_from_creds(m, message, seat_name, error, &seat);
+                if (r < 0)
+                        return r;
 
-        r = manager_get_seat_from_creds(m, message, seat_name, error, &seat);
-        if (r < 0)
-                return r;
+                session = new(Session, 1);
+                if (!session)
+                        return -ENOMEM;
 
-        if (session->seat != seat)
-                return sd_bus_error_setf(error, BUS_ERROR_SESSION_NOT_ON_SEAT,
-                                         "Session %s not on seat %s", session_name, seat_name);
+                *session = (Session) {
+                        .id = strdup(none_id),
+                        .seat = seat,
+                };
```

Comment:
huh, noo... sessions have a life-cycle, allocating a throw-away, unregister object like this is really not an option